MoreLinq.Test/AssertCountTest.cs

```diff
@@ -126,9 +126,16 @@ public void AssertCountWithMismatchingCollectionCount(int sourceCount, int count
             Assert.AreEqual(e.Message, message);
         }
 
-        sealed class BreakingCollection<T> : BreakingSequence<T>, ICollection<T>
+        sealed class BreakingCollection<T> : BreakingCollectionBase<T>, ICollection<T>
         {
-            public BreakingCollection(int count) => Count = count;
+            public BreakingCollection(int count) : base(count)
+            {
+            }
+        }
+
+        abstract class BreakingCollectionBase<T> : BreakingSequence<T>
```

Comment:
The implementation hierarchy feels a little excessive here. Why not keep things separate and simplify to the following?

```diff
diff --git a/MoreLinq.Test/AssertCountTest.cs b/MoreLinq.Test/AssertCountTest.cs
index 02ad8e1..4dc1332 100644
--- a/MoreLinq.Test/AssertCountTest.cs
+++ b/MoreLinq.Test/AssertCountTest.cs
@@ -126,16 +126,9 @@ public void AssertCountWithMismatchingCollectionCount(int sourceCount, int count
             Assert.AreEqual(e.Message, message);
         }
 
-        sealed class BreakingCollection<T> : BreakingCollectionBase<T>, ICollection<T>
+        sealed class BreakingCollection<T> : BreakingSequence<T>, ICollection<T>
         {
-            public BreakingCollection(int count) : base(count)
-            {
-            }
-        }
-
-        abstract class BreakingCollectionBase<T> : BreakingSequence<T>
-        {
-            public BreakingCollectionBase(int count) => Count = count;
+            public BreakingCollection(int count) => Count = count;
 
             public int Count { get; }
 
@@ -155,11 +148,10 @@ public void AssertCountWithReadOnlyCollectionIsLazy()
             new BreakingReadOnlyCollection<object>(5).AssertCount(0);
         }
 
-        sealed class BreakingReadOnlyCollection<T> : BreakingCollectionBase<T>, IReadOnlyCollection<T>
+        sealed class BreakingReadOnlyCollection<T> : BreakingSequence<T>, IReadOnlyCollection<T>
         {
-            public BreakingReadOnlyCollection(int count) : base(count)
-            {
-            }
+            public BreakingReadOnlyCollection(int count) => Count = count;
+            public int Count { get; }
         }
 
 #endif
```